junebug/tests/helpers.py

```diff
@@ -182,10 +182,19 @@ def start_server(self):
         port = reactor.listenTCP(
             0, Site(self.api.app.resource()),
             interface='127.0.0.1')
-        self.addCleanup(port.stopListening)
+        self.service._port = port
+        self.addCleanup(self.stop_server)
         addr = port.getHost()
         self.url = "http://%s:%s" % (addr.host, addr.port)
 
+    @inlineCallbacks
+    def stop_server(self):
+        yield self.service.stopService()
+        for service in self.service:
+            yield self.service.removeService(service)
+        for service in self.service.namedServices.values():
+            yield self.service.removeService(service)
```

Comment:
Ah, then my suggestion to use `removeService` was probbaly a bad one. @jerith Suggestions on a different way to remove the service?